charts/cortex-agent/templates/daemonset.yaml

```diff
@@ -89,10 +89,12 @@ spec:
 
         - name: agent-ids
           mountPath: /etc/traps
-
+        
+        {{- if .Values.agent.createSecret }}
```

Comment:
We need the same `if` statement in `deployment-secret.yaml` file in order to not create the `Secret`

Reply:
Added the ``` if ``` statement in ```deployment-secret.yaml``` as well